### PR TITLE
Allow prebuilt podvm image in e2e test

### LINKED ISSUE #1375: azure: e2e tests should allow prebuilt images
Sometimes it's desirable to skip the build of a podvm image and reference an existing image in a test run. There should be an option for manual tests runs to specify an image-id.

### PR DESCRIPTION
**Pull request by mkulke**
fixes #1375

The podvm-related build steps/jobs will be skipped if the input is empty.

The test provisioner code will use the image id if available, otherwise fall-back to the one built in prior steps.

The test job will hence always run, unless the workflow is cancelled. Drive-by fixed some minor things.

Partially successful (the tests are flaky, atm) run with the 0.7 podvm image [here](https://github.com/mkulke/cloud-api-adaptor/actions/runs/6024476613/job/16343401526)

### CODE REVIEW

File: .github/workflows/azure-e2e-test.yml [hunk 1]
```diff
@@ -216,7 +222,7 @@ jobs:
 
     # Clean up step, run regardless of the failure state.
     - name: Run deprovisioner
-      if: ${{ always() }}
+      if: always()
```
Comment (iaguis) [inline .github/workflows/azure-e2e-test.yml:225]: Nit: I'd do this kind of fixes in a different commit.

File: .github/workflows/azure-e2e-test.yml [hunk 2]
```diff
@@ -261,6 +268,8 @@ jobs:
           --gallery-image-definition ${{ secrets.AZURE_PODVM_IMAGE_DEF_NAME }} \
           --gallery-image-version "${PODVM_IMAGE_VERSION}" || true
 
+    - name: Remove container registry
```
Comment (iaguis) [inline .github/workflows/azure-e2e-test.yml:271]: Nit: it's cleaner to do this adding of names in a different commit.

File: .github/workflows/azure-e2e-test.yml [hunk 3]
```diff
@@ -164,6 +169,7 @@ jobs:
     needs:
     - build-podvm-image
     - build-caa-container-image
+    if: always() && !cancelled()
```
Comment (iaguis) [inline .github/workflows/azure-e2e-test.yml:172]: Maybe I'm misreading the documentation, but given that `always()` returns true always, do we need it here?

**Review by iaguis — COMMENTED** [review]
Some questions.

File: .github/workflows/azure-e2e-test.yml (hunk 2 above)
Comment (mkulke) [reply .github/workflows/azure-e2e-test.yml:271]: The fix is somewhat related, since we still want to remove the container registry, even if we don't want to remove the podvm image. before this maybe not labeled properly but not a problem.

Comment (iaguis) [reply .github/workflows/azure-e2e-test.yml:271]: OK. fair.

File: .github/workflows/azure-e2e-test.yml (hunk 3 above)
Comment (iaguis) [reply .github/workflows/azure-e2e-test.yml:172]: Ah, so this is so this job runs even if `build-podvm-image` is skipped, makes sense.

Let's do what you suggest then.

File: .github/workflows/azure-e2e-test.yml (hunk 2 above)
Comment (surajssd) [reply .github/workflows/azure-e2e-test.yml:271]: This does not delete the entire registry, just the prebuilt image.

**Review by surajssd — APPROVED** [review]

**Review by iaguis — APPROVED** [review]
LGTM if it works :)
